packages/core/src/amazonq/webview/ui/apps/gumbyChatConnector.ts

```diff
@@ -0,0 +1,195 @@
+            tabType: 'gumby',
+            tabOpenInteractionType,
+        })
+    }
+
+    onTabRemove(tabID: string) {
+        this.sendMessageToExtension({
+            tabID: tabID,
+            command: 'tab-was-removed',
+            tabType: 'gumby',
+        })
+    }
+
+    private processChatPrompt = async (messageData: ChatPrompt, tabID: string): Promise<void> => {
+        if (this.onChatAnswerReceived === undefined) {
+            return
+        }
+
+        const answer: ChatItem = {
+            type: ChatItemType.AI_PROMPT,
+            body: messageData.message,
+            formItems: messageData.formItems,
+            buttons: messageData.formButtons,
+            followUp: undefined,
+            status: 'info',
+            canBeVoted: false,
+        }
+
+        this.onChatAnswerReceived(tabID, answer)
+
+        return
+    }
+
+    private processChatMessage = async (messageData: any): Promise<void> => {
+        if (this.onChatAnswerReceived === undefined || this.onChatAnswerUpdated === undefined) {
+            return
+        }
+
+        if (messageData.message !== undefined) {
+            const answer: ChatItem = {
+                type: messageData.messageType,
+                messageId: messageData.messageId ?? messageData.triggerID,
+                body: messageData.message,
+                buttons: messageData.buttons ?? [],
+                canBeVoted: false,
+            }
+
+            if (messageData.messageId !== undefined) {
+                this.onChatAnswerUpdated(messageData.tabID, answer)
+                return
+            }
+
+            this.onChatAnswerReceived(messageData.tabID, answer)
+        }
+    }
+
+    transform = (tabID: string): void => {
+        this.sendMessageToExtension({
+            tabID: tabID,
+            command: 'transform',
+            chatMessage: 'transform',
+            tabType: 'gumby',
+        })
+    }
+
+    private processAuthNeededException = async (messageData: any): Promise<void> => {
+        if (this.onChatAnswerReceived === undefined) {
+            return
+        }
+
+        this.onChatAnswerReceived(messageData.tabID, {
+            type: ChatItemType.SYSTEM_PROMPT,
+            body: messageData.message,
+        })
+    }
+
+    // This handles messages received from the extension, to be forwarded to the webview
+    handleMessageReceive = async (messageData: any): Promise<void> => {
+        if (messageData.type === 'errorMessage') {
+            this.onError(messageData.tabID, messageData.message, messageData.title)
+            return
+        }
+
+        if (messageData.type === 'chatMessage') {
+            await this.processChatMessage(messageData)
+            return
+        }
+
+        if (messageData.type === 'authNeededException') {
+            await this.processAuthNeededException(messageData)
+            return
+        }
+
+        if (messageData.type === 'chatInputEnabledMessage') {
```

Comment:
These `messageData.type` must all be defined somewhere, can we use those?

Reply:
THESE are not -- I will make them so -- thank you for catching